### DISCUSSION TITLE
Using UDP Sockets to Send Commands to Crazyflie

### DISCUSSION THREAD

**Discussion by Mihai-Dev** [bitcraze/discussions#1061] [Q&A]
Hi all,

I'm looking to listen to a UDP socket to accept information to control the Crazyflie. 

The issue here is that the connection works without having commands to the Crazyflie; so for example, just having print statements works fine. However, as soon as I want to trigger a change in the Crazyflie using MotionCommander, the UDP server hangs to the last data it received.

The code can be seen below:

``` python
import socket
import re


import logging
import time
import sys
from threading import Event


import cflib.crtp
from cflib.crazyflie import Crazyflie
from cflib.crazyflie.syncCrazyflie import SyncCrazyflie
from cflib.positioning.motion_commander import MotionCommander
from cflib.utils import uri_helper


UDP_IP = "127.0.0.1"
UDP_PORT = 12345


sock = socket.socket(socket.AF_INET, # Internet
                    socket.SOCK_DGRAM) # UDP
sock.bind((UDP_IP, UDP_PORT))


URI = uri_helper.uri_from_env(default='radio://0/80/2M/E7')
DEFAULT_HEIGHT = 0.6

deck_attached_event = Event()
logging.basicConfig(level=logging.ERROR)
RATE = 360.0 /5

def Assignment_1(scf):
   with MotionCommander(scf, default_height = DEFAULT_HEIGHT) as mc:
       try:
           while True:
             data, addr = sock.recvfrom(1024)
             b = data.decode('utf-8')
             zero_one = re.findall("\d", b)
             print(zero_one)

             if (zero_one == ['1']):
                 print('Taking off...')
                 mc.circle_left(0.1)
             else:
                 print('Landing...')
                 mc.right(0.1)
       except KeyboardInterrupt:
           print('Stopping...')
       finally:
           sock.close()
          
if __name__ == '__main__':
 cflib.crtp.init_drivers()
 with SyncCrazyflie(URI, cf=Crazyflie(rw_cache='./cache')) as scf:
     Assignment_1(scf)
```

I'm unsure why it hangs. The connection is stable - to me it looks like there's an issue with the sending of commands specifically. I don't have much experience with UDP so I am unsure how to solve this.

Any help is appreciated!

---

Edit: @krichardsson you helped out on a similar problem to this; would be great if I could get your input here!

**Comment by krichardsson** [comment]
The calls to `mc.circle_left(0.1)` and 'mc.right(0.1)' are blocking and will not return until they are finished, not sure if this is what you see?

One option is to call the `mc.start_circle_left()` instead that just starts the motion and then call `mc.stop()` later to stop it. See the [implementation of `circle_left()`](https://github.com/bitcraze/crazyflie-lib-python/blob/85f36d3eba42d4cd77ae9205f43060e1c405ebd0/cflib/positioning/motion_commander.py#L232) for instance
